### PR TITLE
Revert "Optimize equality, hashing, serialization."

### PR DESCRIPTION
**Pull request by ormsbee**
This reverts commit a71958ea9b9b4ce1b33dede230bde000caa336e8.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 setup(
     name='edx-opaque-keys',
-    version='0.3.2',  # Please update CHANGELOG when you bump versions
+    version='0.3.1',
```
Comment (doctoryes) [inline setup.py]: The version needs a bump upward...

File: CHANGELOG.rst [hunk 2]
```diff
@@ -1,8 +0,0 @@
-# 0.3.2
-
-* Simple optimizations to reduce the number of OpaqueKey objects
-  created, and to speed up hashing and equality checks.
-
-----
-
-No changelog was maintained before 0.3.2.
```
Comment (doctoryes) [inline CHANGELOG.rst]: We should keep the CHANGELOG and add version 0.3.3?

File: setup.py (hunk 1 above)
Comment (ormsbee) [inline setup.py]: If I bump this right now, this very commit will get uploaded to pypi as 0.3.3. Is that what we want?

Comment (doctoryes) [inline setup.py]: It actually won't until we tag the repo - that's when the automatic Travis pkg build/upload would occur.

Comment (doctoryes) [inline setup.py]: But - perhaps we should just add any additional 0.3.3 changes into this PR to avoid confusion?

Comment (ormsbee) [inline setup.py]: Based on discussion, it sounds like this is indeed what we want.

**Comment by doctoryes** [comment]
👍

**Pull request merged**